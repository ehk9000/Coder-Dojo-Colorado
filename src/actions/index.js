export const addUser = (user) => ({
  type:'ADD_USER',
  user
});

export const isFetching = (bool) => ({
  type: 'IS_FETCHING',
  isFetching: bool
});