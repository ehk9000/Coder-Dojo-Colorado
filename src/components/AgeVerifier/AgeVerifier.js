import React, { Component } from 'react';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

export class AgeVerifier extends Component {
  state = {
    dob: ''
  }

  handleSubmit = (e) => {
    
  } 

  hanldeDate = (date) => {
    this.setState({ dob: date });
  }

  subYears = () => {
    const date = new Date();

    date.setFullYear(date.getFullYear() - 13);
    return date;
  }

  render() {
    const { dob } = this.state;
    return (
      <form className="AgeForm" onSubmit={this.handleSubmit}>
        <h2>Please Verify Your Age</h2>
        <label htmlFor="dob">
          Enter your date of birth
          <DatePicker 
            id="dob"
            onSelect={this.hanldeDate}
            showYearDropdown
            dateFormat="yyyy/MM/dd"
            dropdownMode="select"
            scrollableYearDropdown
            maxDate={this.subYears()}
            showMonthDropdown
            useShortMonthInDropdown
            className="date-picker"
          />
        </label>
        <button type="submit" className="signin-btn">Submit</button>
      </form>
    );
  }
}

export default AgeVerifier;