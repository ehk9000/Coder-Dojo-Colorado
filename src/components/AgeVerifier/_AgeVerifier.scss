.AgeForm {
  @include flex-column;
  margin: 10% auto;
  width:500px;
  min-width: 300px;
  padding: 40px;
  background-color: #FFFFFF;

  h2 {
     margin: 0 0 15px 0;
     text-align: center;
  }
  
  label {
    @include flex-column;
    margin: 0 0 2px 0;
  }
  .signin-btn {
    @include form-submit;
    background-color: #724695;
    border: 2px solid #724695;
  }

  #dob {
    @include form-input-style;
    width: 420px;
    margin-top: 25px;
  }

}
