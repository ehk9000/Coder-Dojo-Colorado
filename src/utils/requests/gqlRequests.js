import { gql } from 'apollo-boost';

export const GET_ALL_USERS = gql`
  {
    allUsers {
      id
      name
      username
      role
      guardianId{
        name
      }
      students{
        name
      }
    }
  }
`;

export const GET_USER_BY_TOKEN = gql`
  {
    me{
        username
        name
        role
    }
  }
`;

export const CREATE_STUDENT = gql`
  mutation createStudent(
    $username: String!,
    $password: String!,
    $name: String!,
    $birthdate: String!    
    ) {
    createStudent(
      username: $username,
      password: $password,
      name: $name,
      birthdate: $birthdate
    ) {
      id
    }
  }
`;

export const CREATE_GUARDIAN = gql`
  mutation createUser(
    $email: String!,
    $username: String!,
    $password: String!,
    $name: String!,
    $phoneNumber: String!,
    $street1: String!,
    $street2: String,
    $city: String!,
    $state: String!,
    $zip: String!
    ) {
    createUser(
      email: $email,
      username: $username,
      password: $password,
      name: $name,
      phoneNumber: $phoneNumber,
      street1: $street1,
      street2: $street2,
      city: $city,
      state: $state,
      zip: $zip
    ) {
      id
    }
  }
`;

export const CREATE_NEW_EVENT = gql`
  mutation createEvent(
    $name: String!,
    $venueId: Int!,
    $notes: String,
    $startTime: String!,
    $endTime: String!
    ) {
    createEvent(
      name: $name,
      venueId: $venueId,
      notes: $notes,
      startTime: $startTime,
      endTime: $endTime
    ) {
      id
    }
  }
`;

export const SIGN_IN = gql`
  mutation signIn(
    $username: String!,
    $password: String!    
    ) {
    signIn(
      username: $username,
      password: $password
    ) {
      token
      user{
        id
        username
        name
        role
      }
    }
  }
`;

export const GET_ALL_VENUES = gql`
  {
    allVenues{
      id
      name
      notes
      addresses{
        city
        street1
        street2
        zip
        state
        id
      }
    }
  }
`;

export const GET_VENUES_BY_CITY = gql`
  query allVenues($city: String!) {
    allVenues(city: $city) {
      id
      name
      notes
      addresses{
        city
        street1
        street2
        zip
        state
        id
      }
    }
  }
`;

export const GET_VENUE_DETAILS = gql`
  query allVenues($venueId: Int!) {
    allVenues(id: $venueId) {
      name
      email
      webUrl
      notes
      addresses {
        street1
        street2
        city
        state
        zip
      }
    }
  }
`;

export const UPCOMING_EVENTS_BY_VENUE = gql`
  query futureEvents($venueId: Int!) {
    futureEvents(venueId: $venueId){
      id
      name
      notes
      startTime
      endTime
    }
  }
`;

export const RECENT_EVENTS_BY_VENUE = gql`
  query pastEvents($venueId: Int!) {
    pastEvents(venueId: $venueId){
      id
      name
      notes
      startTime
      endTime
    }
  }
`;

export const GET_FAMILY = gql`
  {
    me {
      students {
        name
        username
        birthdate
        id
      }
    }
  }
`;

export const GET_ADMIN_DATA = gql`
  {
    me {
      email
      phoneNumber
      username
      addresses{
          street1
          street2
          city
          state
          zip
      }
      venues {
        id
        name
        notes
        email
        webUrl
        events {
          id
          name
          notes
          startTime
          endTime
          eventCode
        }
      }
    }
  }
`;

export const GET_EVENTS_ATTENDED = gql`
 {
    me{
        eventsAttended{
            name
            startTime
            id
        }
    }
}
`;

export const LOG_ATTENDANCE = gql`
  mutation logAttendance( 
    $eventCode: String!
    ) {
    logAttendance(
      eventCode: $eventCode
    ) {
      id
    }
  }
`;

export const GET_GUARDIAN_DATA = gql`
  {
    me {
      email
      phoneNumber
      addresses {
        street1
        street2
        city
        state
        zip
      }
    }
  }
`;
