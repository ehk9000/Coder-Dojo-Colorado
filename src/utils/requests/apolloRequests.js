import ApolloClient from 'apollo-boost';
import store from '../../store';
import { setFetching, setError } from '../../actions';

export function dispatchError(error) {
  const { message } = error.graphQLErrors[0];

  store.dispatch(setError(message));
  store.dispatch(setFetching(false));
}

export function setFetchingAndError() {
  const state = store.getState();
  const { isFetching, error } = state;

  if (!isFetching) store.dispatch(setFetching(true));
  if (error.length) store.dispatch(setError(''));
}

export async function apolloQuery(query, variables) {
  const client = new ApolloClient({
    uri: 'https://cryptic-stream-22818.herokuapp.com/graphql',
    request: (operation) => {
      operation.setContext({
        headers: {
          authorization: localStorage.getItem('token') || ''
        }
      });
    }
  });

  setFetchingAndError();

  let result;

  try {
    result = await client.query({ query, variables });
    if (result) store.dispatch(setFetching(false));
  } catch (error) {
    dispatchError(error);
  }

  return result ? result.data : false;
}

export async function apolloMutate(mutation, variables) {
  const client = new ApolloClient({
    uri: 'https://cryptic-stream-22818.herokuapp.com/graphql',
    request: (operation) => {
      operation.setContext({
        headers: {
          authorization: localStorage.getItem('token') || ''
        }
      });
    }
  });

  setFetchingAndError();

  let result;

  try {
    result = await client.mutate({ mutation, variables });
    if (result) store.dispatch(setFetching(false));
  } catch (error) {
    dispatchError(error);
  }
  
  return result ? result.data : false;
}
