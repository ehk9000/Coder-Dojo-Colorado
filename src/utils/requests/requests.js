import { apolloQuery, apolloMutate } from './apolloRequests';
import * as gql from './gqlRequests';

const requests = {
  signIn: user => apolloMutate(gql.SIGN_IN, user),
  createStudent: student => apolloMutate(gql.CREATE_STUDENT, student),
  createGuardian: guardian => apolloMutate(gql.CREATE_GUARDIAN, guardian),
  getAllUsers: () => apolloQuery(gql.GET_ALL_USERS),
  getUserByToken: () => apolloQuery(gql.GET_USER_BY_TOKEN),
  getAllVenues: () => apolloQuery(gql.GET_ALL_VENUES),
  getVenuesByCity: city => apolloQuery(gql.GET_VENUES_BY_CITY, city),
  getVenueDetails: venueId => apolloQuery(gql.GET_VENUE_DETAILS, venueId),
  getUpcomingEvents: venueId => apolloQuery(gql.UPCOMING_EVENTS_BY_VENUE, venueId),
  getRecentEvents: venueId => apolloQuery(gql.RECENT_EVENTS_BY_VENUE, venueId),
  getAdminDetails: () => apolloQuery(gql.GET_ADMIN_VENUES)
};

export default requests;
