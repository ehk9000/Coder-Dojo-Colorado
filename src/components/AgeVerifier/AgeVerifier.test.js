import React from 'react';
import { shallow } from 'enzyme';
import { AgeVerifier, mapStateToProps, mapDispatchToProps } from './AgeVerifier';
import * as actions from '../../actions';
import requests from '../../utils/requests/requests';

jest.mock('../../actions');
jest.mock('../../utils/requests/requests');

describe('AgeVerifier', () => {
  let wrapper;
  const mockSetError = jest.fn();
  const mockSubmitEvent = {
    preventDefault: jest.fn()
  };

  const mockStudent = {
    fullName: 'Finn',
    username: 'finn1',
    password: 'password'
  };

  beforeEach(() => {
    wrapper = shallow(
      <AgeVerifier
        setError={mockSetError}
        error=""
        loading={false}
        mockStudent={mockStudent}
      />
    );
  });
  
  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      birthdate: '',
      success: false
    });
  });

  it('should be able to handle submit', () => {
    wrapper.instance().handleSubmit(mockSubmitEvent);
    expect(mockSubmitEvent.preventDefault).toHaveBeenCalled();
  });

  it('should create a student in the db ', async () => {
    wrapper.setState({
      birthdate: '08/17/19',
      successs: false
    });

    await wrapper.instance().createStudent();

    expect(requests.createStudent).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with setError', () => {
      const error = 'Something went wrong';
      const action = actions.setError(error);
      const dispatch = jest.fn();

      mapDispatchToProps(dispatch).setError(error);

      expect(dispatch).toHaveBeenCalledWith(action)
    });
    it('should call addUser with a student', () => {
      const action = actions.addUser(mockStudent);
      const dispatch = jest.fn();

      mapDispatchToProps(dispatch).addUser(mockStudent);

      expect(dispatch).toHaveBeenCalledWith(action);
    });
  });
  describe('mapStateToProps', () => {
    it('should MSTP', () => {
      const state = {
        error: '',
        loading: false
      };

      const { error } = mapStateToProps(state);

      expect(error).toEqual(state.error);
    });
  }); 
});
