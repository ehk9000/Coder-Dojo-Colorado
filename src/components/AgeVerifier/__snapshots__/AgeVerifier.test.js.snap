// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgeVerifier should match snapshot 1`] = `
<form
  className="AgeForm"
  onSubmit={[Function]}
>
  <h2>
    Please Verify Your Age
  </h2>
  <label
    htmlFor="dob"
  >
    Enter your date of birth
    <DatePicker
      allowSameDay={false}
      className="date-picker"
      dateFormat="yyyy/MM/dd"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="select"
      id="dob"
      inlineFocusSelectedMonth={false}
      maxDate={2006-07-20T19:32:48.382Z}
      monthsShown={1}
      nextMonthButtonLabel="Next month"
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      readOnly={false}
      renderDayContents={[Function]}
      scrollableYearDropdown={true}
      shouldCloseOnSelect={true}
      showMonthDropdown={true}
      showMonthYearPicker={false}
      showTimeInput={false}
      showTimeSelect={false}
      showYearDropdown={true}
      strictParsing={false}
      timeCaption="Time"
      timeInputLabel="Time"
      timeIntervals={30}
      useShortMonthInDropdown={true}
      withPortal={false}
    />
  </label>
  <button
    className="signin-btn"
    type="submit"
  >
    Submit
  </button>
</form>
`;
